src/praisonai-agents/praisonaiagents/eval/trials.py

```
@@ -0,0 +1,445 @@
+
+
+def _isolated_agent_copy(agent: Any) -> Any:
+    """Return a best-effort isolated copy of the agent for one attempt.
+
+    Each attempt should be an independent sample: fresh session/chat history,
+    caches cleared, and memory/knowledge *writes* severed so the caller's stores
+    are never mutated. Falls back to the original agent when copy is unsupported.
+    """
+    try:
+        clone = copy.copy(agent)
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)